### ISSUE TITLE
Firefox version

### ISSUE THREAD

**Issue by yoasif** [leggett/simplify#1]
It'd be nice to make this work as a WebExtension for Firefox users. 

This page may be helpful: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Porting_a_Google_Chrome_extension

**Comment by nathancahill** [comment]
I volunteer to contribute a port if the PR would be accepted. I'd like to know beforehand though.

**Comment by albizures** [comment]
Hey @nathancahill, me too ✋, I would love help you if you need to

**Comment by Skraelingjar** [comment]
I'm happy to help maintain a port after it's working, if needed.

**Comment by theFroh** [comment]
I think it already does. I cloned the repo, popped open `about:debugging` and tested via "Load Temporary Add-on" by picking the manifest file. From what I can tell at a glance my G-Mail is sufficiently simplified.

That said, I don't know if it is doing "everything".

**Comment by omgmog** [comment]
You could just take the CSS from this repo, change all of the instances of `html.simpl` to `html` and use the CSS with Stylish/Stylus on Firefox.

**Comment by leggett** [comment]
Love that Firefox support is the first request. 👍 Happy to help getting this working in Firefox but not wanting to maintain that at this moment. 

There are three html-level classnames:

1. html.simpl - the global classname for toggling the styles on/off (with Command+J). Added on page load. 
2. html.inSettings - apply changes when you are settings. Added when you go to Settings.
3. html.hideSearch - minimize search to an icon (removed when you expand Search). Added as soon as search is loaded (bad things happen if it is applied before)

I don't understand the problem though. You should be able to add classnames on page load in Firefox too, no?

**Comment by nathancahill** [comment]
Not sure what you mean about maintaining it, the Webextension API is cross-browser so the only additional thing you'd have to do is publish new versions to the Firefox Addons site the same way you publish new versions to the Chrome Web Store.

**Comment by leggett** [comment]
I mean that Firefox uses a different rendering engine so any changes I have to make as Gmail changes, I'd have to test twice in all the various configs and screens. It shouldn't be different, but it could be and I wouldn't want to do it myself unless I was willing to take on that additional testing with each update. There are also possibly CSS features I'm using that aren't supported by Firefox which would mean I'd need to find a different way to adjust that thing. If I can get to a place where things are more stable, I will look into it.

**Comment by leggett** [comment]
Let me look into it before I respond further. I agree that having another extension that isn't getting updated but is under the same name sounds bad.

**Comment by leggett** [comment]
More guidelines here: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/Distribution/Submitting_an_add-on

**Comment by chipironcin** [comment]
I checked the extension with the official tool at https://www.extensiontest.com/ and it returned no compatibility issues at all. So maybe direct publishing will (just work)tm

**Comment by wolffc** [comment]
I just downloades your sourcecode and "enabled" the extension successfully in Firefox Developer Editon 67.0b13 (64-bit).

so i guess the only thing you have to do ist to publish your extenso to firefox users

**Comment by Akryum** [comment]
🙏

**Comment by leggett** [comment]
@wolffc I'd want to test a good sample of all the different configurations to make sure all my overrides are working correctly. 

Specifically, I'd want to test the following at three different window widths: small (< 960px), medium (961 - 1440px) and large (> 1440px):

- The 6 different inbox set-ups in default view
- Preview pane (vertical and horizontal) in default inbox and priority inbox
- Multiple Inboxes
- Right side chat
- Offline (causes weird problems sometimes)
- Light, dark and photo theme
- Add-ons opening and closing, and does the compose button move when you open/close it
- Nav showing / hiding (and make sure nothing odd happens when you hover over the compose button which is technically in the nav still)
- Search minimized, expanded, and does closing a search go back to the previous view
- Does changing the inbox density get respected on reload
- Does pagination hide if the buttons are not active

**Comment by dmose** [comment]
Regarding the extra testing matrix, I wonder if it would be helpful/practical to automate rendering of each screen in a harness (eg [Storybook](@storybook.js.org), and perhaps additionally use a [automated visual testing system](https://storybook.js.org/docs/testing/automated-visual-testing/) to make things even easier.

**Comment by jonathan-s** [comment]
If you find that Storybook is not a good fit browser testing can also be done in cypress. https://docs.cypress.io/guides/tooling/plugins-guide.html#Browser-Launching

**Comment by chrispickard** [comment]
I was also able to install this on firefox 66 and it seems to work great, thanks!

It took me a second to figure out how to package it, so...
```
git clone git@github.com:leggett/simplify.git
cd simplify/gmail
zip -r -FS ../simplify.zip *
```
then open `about:addons`
click the settings menu
select "Debug Add-ons"
click "Load Temporary Add-on"
select the `simplify.zip` file you created

**Comment by mstfacmly** [comment]
Just to add my voice here: I added the manifest.json through the temporary add-onds menu, and the difference it makes to GMail's interface is astounding.

I hope the above news means we'll be seeing an official release on the add-ons page.

**Comment by vikrantt** [comment]
This works for Firefox. I have just uploaded it. Needs improvements though. You can find it here: https://addons.mozilla.org/en-US/firefox/addon/simplify-gmail/

**Comment by YurySolovyov** [comment]
Why not wait for the author to do that?

**Comment by rajeevedmonds** [comment]
Exactly! I'll prefer the original author to publish it.

**Comment by vikrantt** [comment]
Guys

Because he said he’s busy and encouraged others. I was testing it for my use and then thought others could use it too. When he’d have time to publish, I can remove this one. I linked this github page anyways and gave him credit. So don’t understand the issue.

**Comment by TheFlatworm** [comment]
I, for one, am thankful!

**Comment by ciex** [comment]
@vikrantt The only issue I see is that you claim an MIT license where leggett didn't specify any license as far as I can tell.

**Comment by vikrantt** [comment]
If only you had bothered to look in the code before commenting :

Look in gmail/LICENSE.md

**Comment by leggett** [comment]
Love the enthusasm and the help. Thanks. 

@vikrantt I'm working on the firefox port -- can you give me a little more time (at least the weekend). I don't think it will be a good experience for people to add this and then have it go away.

**Comment by doliG** [comment]
Just here to say thanks to @leggett for this extension, and to @vikrantt for the firefox port.
You rocks guys :)

**Comment by rajeevedmonds** [comment]
@doliG There's no official Firefox port as of yet. If you're looking forward to a stable, thoroughly tested, and diligently maintained port, I'll suggest waiting for the official version published by the original author.

**Comment by vikrantt** [comment]
> Love the enthusiasm and the help. Thanks.
> 
> @vikrantt I'm working on the firefox port -- can you give me a little more time (at least the weekend). I don't think it will be a good experience for people to add this and then have it go away.

@leggett  sorry for confusion man; I didn't know you were porting. This is spiraling out of control. Could you give me your email? I will just add it to my uploaded AddOn on Mozilla Network as the Owner and eventually remove myself. That way you will have full control and can remove/add updates in future. I have sent you a request on LinkedIn so that we can do the email-info exchange in a private conversation.

**Comment by leggett** [comment]
No worries. I should have just published it on Firefox as many were encouraging from the get go. 
You can email me at hi.simplify@gmail.com. It is probably easiest if you just take your version down but happy to discuss other options if you think it is better.

**Comment by leggett** [comment]
So that was easy (as many said it would be). 

I published it here: https://addons.mozilla.org/en-US/firefox/addon/simplifygmail/ and I created a short-url at https://firefox.simpl.fyi that points to that URL. There is at least one issue but it is functional and I can iterate on it as I'm doing with the Chrome extension.

Now, I have to figure out what to do with the duplicates created because I didn't get this out sooner:
https://addons.mozilla.org/en-US/firefox/addon/simplify-gmail/
https://addons.mozilla.org/en-US/firefox/addon/simpl-fyi/

Any chance any of those are by people on this issue?
